lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java

```diff
@@ -912,32 +913,6 @@ public Drawable getContentBackground() {
     return getImpl().getContentBackground();
   }
 
-  private static int resolveAdjustedSize(int desiredSize, int measureSpec) {
-    int result = desiredSize;
-    int specMode = MeasureSpec.getMode(measureSpec);
-    int specSize = MeasureSpec.getSize(measureSpec);
-    switch (specMode) {
-      case MeasureSpec.UNSPECIFIED:
-        // Parent says we can be as big as we want. Just don't be larger
-        // than max size imposed on ourselves.
-        result = desiredSize;
-        break;
-      case MeasureSpec.AT_MOST:
-        // Parent says we can be as big as we want, up to specSize.
-        // Don't be larger than specSize, and don't be larger than
-        // the max size imposed on ourselves.
-        result = Math.min(desiredSize, specSize);
```

Comment:
The behavior seems slightly different here? TBH I'm not clear why this method was created at the very beginning. Let me ask some people who may have more knowledge about this.

Reply:
Yes, it is, but for all acceptable arguments, `View.resolveSize` will return the same values.

Reply:
Ah you are right. MEASURED_STATE_TOO_SMALL will be removed by View.resolveSize()